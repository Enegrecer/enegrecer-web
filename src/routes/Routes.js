import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import HomePage from '../components/home/Home';
import Navbar from '../components/navbar/Navbar';
import Rodape from '../components/rodape/Rodape';
import Sobre from '../components/sobrePortal/Sobre';
import ColetivoEnegrecer from '../components/sobreColetivo/ColetivoEnegrecer';
import NovaDenunciaContainer from '../containers/denuncias/NovaDenunciaContainer';
import DepoisDaDenuncia from '../components/depoisdadenuncia/DepoisDaDenuncia';
import ModeradorLogin from '../components/moderador/ModeradorLogin';
import PaginaNaoEncontrada from '../components/paginaNaoEncontrada/PaginaNaoEncontrada';
import PainelModerador from '../components/moderador/painelModerador';
import MenuModerador from '../components/moderador/menuModerador';
import { estaAutenticado } from '../utils/firebaseUtils';
import PrivateRoute from './PrivateRoute';
import VisualizarDenunciaContainer from '../containers/denuncias/VisualizarDenunciaContainer';

const Routes = () => (
  <BrowserRouter>
    <div>
      {estaAutenticado() ? <MenuModerador /> : <Navbar />}
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/sobre" component={Sobre} />
        <Route exact path="/coletivo-enegrecer" component={ColetivoEnegrecer} />
        <Route exact path="/denunciar" component={NovaDenunciaContainer} />
        <Route exact path="/depois-denuncia" component={DepoisDaDenuncia} />
        <Route exact path="/login" component={ModeradorLogin} />
        <Route exact path="/moderador/visualizar-denuncia/:denunciaId" component={VisualizarDenunciaContainer} />
        <PrivateRoute path="/moderador" component={PainelModerador} />

        <Route component={PaginaNaoEncontrada} />
      </Switch>
      <Rodape />
    </div>
  </BrowserRouter>
);

export default Routes;
