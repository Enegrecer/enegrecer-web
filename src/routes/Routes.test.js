import React from 'react';
import { shallow } from 'enzyme';
import Routes from './Routes';
import HomePage from '../components/home/Home';
import NovaDenuncia from '../containers/denuncias/NovaDenunciaContainer';


function possuiRotaComComponente(rotas, rota, componente) {
  it(`possui rota "${rota}" com componente correto`, () => {
    const seletor = `[path="${rota}"]`;
    expect(rotas.find(seletor).prop('component')).toEqual(componente);
  });
}

describe('Routes', () => {
  const routes = shallow(<Routes />);
  possuiRotaComComponente(routes, '/', HomePage);
  possuiRotaComComponente(routes, '/painel/denuncias/nova', NovaDenuncia);
});

