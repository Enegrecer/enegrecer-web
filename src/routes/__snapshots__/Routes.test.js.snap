// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes com moderador logado deve construir a pagina com menu para moderador 1`] = `
<BrowserRouter>
  <div>
    <Menu />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/sobre"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/coletivo-enegrecer"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/denunciar"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/depois-denuncia"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/moderador/visualizar-denuncia/:denunciaId"
      />
      <PrivateRoute
        component={[Function]}
        path="/moderador"
      />
      <Route
        component={[Function]}
      />
    </Switch>
    <rodape />
  </div>
</BrowserRouter>
`;

exports[`Routes sem moderador logado deve construir a pagina com menu da p√°gina 1`] = `
<BrowserRouter>
  <div>
    <Menu />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/sobre"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/coletivo-enegrecer"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/denunciar"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/depois-denuncia"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/moderador/visualizar-denuncia/:denunciaId"
      />
      <PrivateRoute
        component={[Function]}
        path="/moderador"
      />
      <Route
        component={[Function]}
      />
    </Switch>
    <rodape />
  </div>
</BrowserRouter>
`;
